lib/trino-parquet/src/main/java/io/trino/parquet/writer/PrimitiveColumnWriter.java

```diff
@@ -241,11 +242,13 @@ private void flushCurrentPageToBuffer()
 
         // update total stats
         totalUnCompressedSize += pageHeader.size() + uncompressedSize;
-        totalCompressedSize += pageHeader.size() + compressedSize;
+        long pageCompressedSize = pageHeader.size() + compressedSize;
+        totalCompressedSize += pageCompressedSize;
         totalValues += valueCount;
 
         pageBuffer.add(pageHeader);
         pageBuffer.add(pageData);
+        pageBufferedBytes += pageCompressedSize;
```

Comment:
`pageBufferedBytes` never goes down now

Reply:
It wasn't going down earlier as well because nothing is removed from `pageBuffer` list